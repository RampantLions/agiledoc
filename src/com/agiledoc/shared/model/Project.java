package com.agiledoc.shared.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Project implements Serializable {

	private String name;

	private String root;

	private String domain;

	private String featureFullPackage;

	private String featureRelativePackage;

	private Classe[] classes;

	private Classe[] features;

	private List<TodoClasse> todoList;

	private ArrayList<Productivity> progressiveProductivity;

	private Productivity cumulativeProductivity;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getRoot() {
		return root;
	}

	public void setRoot(String root) {
		this.root = root;
	}

	public String getDomain() {
		return domain;
	}

	public void setDomain(String domain) {
		this.domain = domain;
		this.featureFullPackage = domain + "." + featureRelativePackage;
	}

	public String getFeatureFullPackage() {
		return featureFullPackage;
	}

	public String getFeatureRelativePackage() {
		return featureRelativePackage;
	}

	public void setFeatureRelativePackage(String featureRelativePackage) {
		this.featureRelativePackage = featureRelativePackage;
		this.featureFullPackage = domain + "." + featureRelativePackage;
	}

	public Classe[] getClasses() {
		return classes;
	}

	public void setClasses(Classe[] classes) {
		this.classes = classes;
	}

	public ArrayList<Productivity> getProgressiveProductivity() {
		return progressiveProductivity;
	}

	public void setProgressiveProductivity(
			ArrayList<Productivity> progressiveProductivity) {
		this.progressiveProductivity = progressiveProductivity;
	}

	public Productivity getCumulativeProductivity() {
		return cumulativeProductivity;
	}

	public void setCumulativeProductivity(Productivity cumulativeProductivity) {
		this.cumulativeProductivity = cumulativeProductivity;
	}

	public Classe[] getFeatures() {
		return features;
	}

	public void setFeatures(Classe[] features) {
		this.features = features;
	}

	public List<TodoClasse> getTodoList() {
		return todoList;
	}

	public void setTodoList(List<TodoClasse> todoList) {
		this.todoList = todoList;
	}

}
