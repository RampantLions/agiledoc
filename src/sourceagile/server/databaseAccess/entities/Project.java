package sourceagile.server.databaseAccess.entities;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import javax.jdo.annotations.Column;
import javax.jdo.annotations.IdGeneratorStrategy;
import javax.jdo.annotations.PersistenceCapable;
import javax.jdo.annotations.Persistent;
import javax.jdo.annotations.PrimaryKey;

import com.google.appengine.api.datastore.Key;

@PersistenceCapable
public class Project implements Serializable {

	private static final long serialVersionUID = 1L;

	@PrimaryKey
	@Persistent(valueStrategy = IdGeneratorStrategy.IDENTITY)
	private Key projectID;

	@Persistent
	private String name;

	@Persistent
	@Column(jdbcType = "LONGVARCHAR")
	private String description;

	@Persistent
	private String repositoryType;

	@Persistent
	private String repositoryURL;

	@Persistent
	private String root;

	@Persistent
	private String testRoot;

	@Persistent
	private String domain;

	@Persistent
	@Column(jdbcType = "LONGVARCHAR")
	private String relatedLinks;

	@Persistent(defaultFetchGroup = "true")
	private List<ProjectComponents> projectComponents = new ArrayList<ProjectComponents>();

	@Persistent(defaultFetchGroup = "true")
	private List<ProjectBacklog> projectBacklog = new ArrayList<ProjectBacklog>();

	public Key getProjectID() {
		return projectID;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getRepositoryType() {
		return repositoryType;
	}

	public void setRepositoryType(String repositoryType) {
		this.repositoryType = repositoryType;
	}

	public String getRoot() {
		return root;
	}

	public void setRoot(String root) {
		this.root = root;
	}

	public String getDomain() {
		return domain;
	}

	public String getRepositoryURL() {
		return repositoryURL;
	}

	public void setRepositoryURL(String repositoryURL) {
		this.repositoryURL = repositoryURL;
	}

	public void setDomain(String domain) {
		this.domain = domain;
	}

	public String getTestRoot() {
		return testRoot;
	}

	public void setTestRoot(String testRoot) {
		this.testRoot = testRoot;
	}

	public String getRelatedLinks() {
		return relatedLinks;
	}

	public void setRelatedLinks(String relatedLinks) {
		this.relatedLinks = relatedLinks;
	}

	public List<ProjectComponents> getProjectComponents() {
		return projectComponents;
	}

	public void setProjectComponents(List<ProjectComponents> projectComponents) {
		this.projectComponents = projectComponents;
	}

	public List<ProjectBacklog> getProjectBacklog() {
		return projectBacklog;
	}

	public void setProjectBacklog(List<ProjectBacklog> projectBacklog) {
		this.projectBacklog = projectBacklog;
	}

}
