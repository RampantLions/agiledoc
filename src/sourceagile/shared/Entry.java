package sourceagile.shared;

import java.io.Serializable;
import java.util.Date;

public class Entry implements Serializable, Comparable<Entry> {

	private static final long serialVersionUID = 1L;

	private String classPath;

	private String className;

	private Feature feature;

	private ClassDocumentation classDoc;

	private Date dateModified;

	private User user;

	private String textContent;

	public String getClassPath() {
		return classPath;
	}

	public void setClassPath(String classPath) {
		this.classPath = classPath;
	}

	public String getClassName() {
		return className;
	}

	public void setClassName(String className) {
		this.className = className;
	}

	public Date getDateModified() {
		return dateModified;
	}

	public void setDateModified(Date dateModified) {
		this.dateModified = dateModified;
	}

	public Feature getFeature() {
		return feature;
	}

	public void setFeature(Feature feature) {
		this.feature = feature;
	}

	public String getTextContent() {
		return textContent;
	}

	public void setTextContent(String textContent) {
		this.textContent = textContent;
	}

	public ClassDocumentation getClassDoc() {
		return classDoc;
	}

	public void setClassDoc(ClassDocumentation classDoc) {
		this.classDoc = classDoc;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	@Override
	public int compareTo(Entry that) {

		String thisClass = this.classPath + "/" + this.className;
		String thatClass = that.classPath + "/" + that.className;

		return thisClass.compareTo(thatClass);
	}

	@Override
	public String toString() {

		if (this.classPath == null || this.classPath.length() == 0) {

			return this.className;
		} else {

			return this.classPath + "/" + this.className;
		}
	}

}
