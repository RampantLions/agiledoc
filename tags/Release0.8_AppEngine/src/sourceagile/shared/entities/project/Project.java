package sourceagile.shared.entities.project;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * 
 * @Entity
 */
public class Project implements Serializable {

	private static final long serialVersionUID = 1L;

	public static final String REPOSITORY_TYPE_SUBVERSION = "Subversion";
	public static final String REPOSITORY_TYPE_GIT = "Git";
	public static final String REPOSITORY_TYPE_XML = "XML";

	private Long projectID;

	private String name;

	private String description;

	private String repositoryType;

	private String repositoryURL;

	private String root;

	private String testRoot;

	private String domain;

	private String relatedLinks;

	private List<ProjectComponents> projectComponents = new ArrayList<ProjectComponents>();

	private List<ProjectBacklog> projectBacklog = new ArrayList<ProjectBacklog>();

	public Long getProjectID() {
		return projectID;
	}

	public void setProjectID(Long projectID) {
		this.projectID = projectID;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getRepositoryType() {
		return repositoryType;
	}

	public void setRepositoryType(String repositoryType) {
		this.repositoryType = repositoryType;
	}

	public String getRoot() {
		return root;
	}

	public void setRoot(String root) {
		this.root = root;
	}

	public String getDomain() {
		return domain;
	}

	public String getRepositoryURL() {
		return repositoryURL;
	}

	public void setRepositoryURL(String repositoryURL) {
		this.repositoryURL = repositoryURL;
	}

	public void setDomain(String domain) {
		this.domain = domain;
	}

	public String getTestRoot() {
		return testRoot;
	}

	public void setTestRoot(String testRoot) {
		this.testRoot = testRoot;
	}

	public String getRelatedLinks() {
		return relatedLinks;
	}

	public void setRelatedLinks(String relatedLinks) {
		this.relatedLinks = relatedLinks;
	}

	public List<ProjectComponents> getProjectComponents() {
		return projectComponents;
	}

	public void setProjectComponents(List<ProjectComponents> projectComponents) {
		this.projectComponents = projectComponents;
	}

	public List<ProjectBacklog> getProjectBacklog() {
		return projectBacklog;
	}

	public void setProjectBacklog(List<ProjectBacklog> projectBacklog) {
		this.projectBacklog = projectBacklog;
	}

}
