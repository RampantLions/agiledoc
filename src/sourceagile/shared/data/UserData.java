package sourceagile.shared.data;

import sourceagile.shared.entities.User;

public class UserData {

	public static User[] load() {

		User[] users = new User[3];

		users[0] = new User();
		users[0].setNickName("Alline Oliveira");
		users[0].setName("Alline.Oliveira");
		users[0].setPassword("bD8Km8se2rV9");

		users[1] = new User();
		users[1].setNickName("June Clarke");
		users[1].setName("joonspoon");
		users[1].setPassword("Vj6mz4MX9Wu7");

		users[2] = new User();
		users[2].setNickName("Alline Oliveira");
		users[2].setName("alline");
		users[2].setPassword("agile");

		return users;
	}
}
