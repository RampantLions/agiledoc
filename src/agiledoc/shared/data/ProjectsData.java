package agiledoc.shared.data;

import agiledoc.shared.Project;

public class ProjectsData {

	public static Project[] load() {

		Project[] projs = new Project[1];

		projs[0] = new Project();
		projs[0].setName("AgileDoc");
		projs[0].setRepositoryURL("http://agiledoc.googlecode.com");
		projs[0].setRoot("/svn/trunk/src/");
		projs[0].setDomain("agiledoc/client");

		return projs;
	}
}
