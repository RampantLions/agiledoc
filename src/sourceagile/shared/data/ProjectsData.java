package sourceagile.shared.data;

import sourceagile.shared.Project;

public class ProjectsData {

	public static Project[] load() {

		Project[] projs = new Project[2];

		projs[0] = new Project();
		projs[0].setName("Source Agile");
		projs[0].setRepositoryURL("http://agiledoc.googlecode.com");
		projs[0].setRoot("/svn/trunk/src/");
		projs[0].setDomain("sourceagile/client");

		projs[1] = new Project();
		projs[1].setName("Tic Tac Toe");
		projs[1].setRepositoryURL("http://tictactoe4j.googlecode.com");
		projs[1].setRoot("/svn/trunk/src/");
		projs[1].setDomain("tictactoe/client");

		return projs;
	}
}
