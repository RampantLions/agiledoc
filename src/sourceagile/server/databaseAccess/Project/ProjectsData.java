package sourceagile.server.databaseAccess.Project;

import java.util.ArrayList;
import java.util.List;

import sourceagile.shared.entities.project.Project;
import sourceagile.shared.entities.project.ProjectComponents;

public class ProjectsData {

	public static Project[] load() {

		Project[] projs = new Project[3];

		// loadProject0(projs);
		// loadProject1(projs);
		// loadProject2(projs);

		return projs;
	}

	public static Project loadProject0() {

		Project project = new Project();
		// project.setProjectID(new Long(1));
		project.setName("Source Agile");
		project.setRepositoryType(sourceagile.shared.entities.project.Project.REPOSITORY_TYPE_SUBVERSION);
		project.setRepositoryURL("https://agiledoc.googlecode.com/svn");
		project.setRoot("/trunk/src/");
		project.setTestRoot("/trunk/test/");
		project.setDomain("sourceagile");
		project.setRelatedLinks("http://code.google.com/p/agiledoc/wiki/SourceAgile");

		List<ProjectComponents> projectComponents = new ArrayList<ProjectComponents>();

		projectComponents.add(new ProjectComponents("Authentication",
				"authentication/client"));
		projectComponents.add(new ProjectComponents("System Navigation",
				"client/systemNavigation"));
		projectComponents.add(new ProjectComponents("Planning",
				"planning/client"));
		projectComponents.add(new ProjectComponents("Development",
				"development/client"));
		projectComponents.add(new ProjectComponents("Documentation",
				"documentation/client"));

		projectComponents
				.add(new ProjectComponents("Testing", "testing/client"));
		projectComponents.add(new ProjectComponents("Code Visualization",
				"codeview/client"));
		projectComponents
				.add(new ProjectComponents("Metrics", "metrics/client"));

		projectComponents.add(new ProjectComponents(
				"Infra Structure/Class Repositories",
				"server/classRepositories"));
		projectComponents.add(new ProjectComponents(
				"Infra Structure/Database Access", "server/databaseAccess"));
		projectComponents.add(new ProjectComponents("Infra Structure/Doclet",
				"server/doclet"));

		projectComponents.add(new ProjectComponents("Entities",
				"shared/entities"));

		project.setProjectComponents(projectComponents);

		return project;
	}

	public static Project loadProject1() {

		Project project = new Project();
		// project.setProjectID(new Long(2));
		project.setName("Tic Tac Toe");
		project.setRepositoryType(sourceagile.shared.entities.project.Project.REPOSITORY_TYPE_SUBVERSION);
		project.setRepositoryURL("https://tictactoe4j.googlecode.com/svn/trunk");
		project.setRoot("/src/");
		project.setTestRoot("/test/");
		project.setDomain("tictactoe");
		project.setRelatedLinks("http://en.wikipedia.org/wiki/Tic-tac-toe");
		project.setDescription("Tic-tac-toe, also spelled tick tack toe, or noughts and crosses as it is known in the UK, Ireland, "
				+ "Australia, New Zealand, is a pencil-and-paper game for two players, O and X, who take turns marking the spaces in a 3x3 grid,"
				+ " usually X going first. The player who succeeds in placing three respective marks in a horizontal, vertical, or diagonal row wins the game.");

		project.setRelatedLinks("Tic Tac Toe online version:"
				+ "(Google App Engine): http://tictactoe4j.appspot.com"
				+ "What is Tic Tac Toe: "
				+ "http://en.wikipedia.org/wiki/Tic-tac-toe"
				+ "tictactoe4j  -  File Repository:"
				+ "Google Code - Subversion:"
				+ "# Non-members may check out a read-only working copy anonymously over HTTP."
				+ "svn checkout http://tictactoe4j.googlecode.com/svn/trunk/ tictactoe4j-read-only");

		List<ProjectComponents> projectComponents = new ArrayList<ProjectComponents>();

		projectComponents.add(new ProjectComponents("Authentication",
				"src/tictactoe/authentication/client"));
		projectComponents.add(new ProjectComponents("Gaming",
				"src/tictactoe/client"));
		projectComponents.add(new ProjectComponents("Infra Structure/Server",
				"src/tictactoe/server"));
		projectComponents.add(new ProjectComponents("Infra Structure/Entities",
				"src/tictactoe/shared"));

		project.setProjectComponents(projectComponents);

		return project;

	}

	public static Project loadProject2() {

		Project project = new Project();
		// project.setProjectID(new Long(3));
		project.setName("Help Agile");
		project.setRepositoryType(sourceagile.shared.entities.project.Project.REPOSITORY_TYPE_SUBVERSION);
		project.setRepositoryURL("https://agilehelp.googlecode.com/svn");
		project.setRoot("/trunk/src/");
		project.setTestRoot("/trunk/test/");
		project.setDomain("helpagile");
		project.setDescription("Help Agile is an automatic generated specification "
				+ "and user documentation for your software.");

		List<ProjectComponents> projectComponents = new ArrayList<ProjectComponents>();

		projectComponents.add(new ProjectComponents("Help Agile", "client"));

		project.setProjectComponents(projectComponents);

		return project;
	}

	public static Project loadProject3() {

		Project project = new Project();
		// project.setProjectID(new Long(3));
		project.setName("Where is URL?");
		project.setRepositoryType(sourceagile.shared.entities.project.Project.REPOSITORY_TYPE_SUBVERSION);
		project.setRepositoryURL("http://platrasha.svnrepository.com/svn/code");
		project.setRoot("/client/scheduler/src/com/platrasha/trip/client/scheduler/client/");
		project.setTestRoot("/client/scheduler/test/com/platrasha/trip/client/scheduler/client/");
		project.setDomain("daywidget");
		project.setDescription("");

		List<ProjectComponents> projectComponents = new ArrayList<ProjectComponents>();

		projectComponents.add(new ProjectComponents("Date", "date"));
		projectComponents.add(new ProjectComponents("Editor", "editor"));
		projectComponents.add(new ProjectComponents("Events", "events"));
		projectComponents.add(new ProjectComponents("Panels", "panels"));
		projectComponents.add(new ProjectComponents("Table", "table"));
		projectComponents.add(new ProjectComponents("Utils", "utils"));

		project.setProjectComponents(projectComponents);

		return project;
	}

	public static Project loadProject4() {

		Project proj = new Project();
		// proj.setProjectID(new Long(4));
		proj.setName("Jogo Da Velha");
		proj.setProjectLocale(Project.LOCALE_PORTUGUESE);
		proj.setRepositoryType(Project.REPOSITORY_TYPE_XML);
		proj.setRepositoryURL("https://tictactoe4j.googlecode.com/svn");
		proj.setRoot("/tags/Release_0.7_JogoDaVelha/src/");
		proj.setTestRoot("/tags/Release_0.7_JogoDaVelha/test/");
		proj.setDomain("jogoDaVelha");
		proj.setRelatedLinks("http://pt.wikipedia.org/wiki/Jogo_da_velha");

		proj.setDescription("O Jogo da Velha (português brasileiro) ou jogo do galo (português europeu) "
				+ "é um jogo e passatempo popular."
				+ " Um jogo de lápis e papel para dois jogadores, O e X, que se revezam marcando os espaços "
				+ " em uma grade 3x3. O jogador que conseguir colocar três respectivas marcas em uma linha horizontal,"
				+ " vertical ou diagonal, ganha o jogo.");

		List<ProjectComponents> projectComponents = new ArrayList<ProjectComponents>();

		projectComponents.add(new ProjectComponents("Autenticação",
				"autenticacao/client"));
		projectComponents.add(new ProjectComponents("Jogo", "client"));
		projectComponents.add(new ProjectComponents("Infra-estrutura/Servidor",
				"server"));
		projectComponents.add(new ProjectComponents(
				"Infra-estrutura/Entidades", "shared"));

		proj.setProjectComponents(projectComponents);

		return proj;
	}

	public static Project loadProject5() {

		Project project = new Project();
		// project.setProjectID(new Long(2));
		project.setName("TravelLog");
		project.setProjectLocale(Project.LOCALE_ENGLISH);
		project.setRepositoryType(sourceagile.shared.entities.project.Project.REPOSITORY_TYPE_XML);
		project.setRepositoryURL("https://travellog-aws.googlecode.com/svn/TravelLog");
		project.setRoot("/src/");
		project.setTestRoot("/test/");
		project.setDomain("com.amazon.aws.samplecode.travellog");

		project.setDescription("A Java Web Application demostration the use of Amazon S3, Amazon "
				+ "SimpleDB and Amazon SNS. "
				+ "Share unlimited photos and videos, and stay in touch while you travel."
				+ " A travel journal contains descriptions of the traveler's experiences, "
				+ " is normally written during the course of the journey, intended on "
				+ " updating friends or family on the journey.");

		project.setRelatedLinks("TravelLog online version:"
				+ "http:// "
				+ "What is a Travel Log: "
				+ "http://en.wikipedia.org/wiki/Travel_journal"
				+ "tictactoe4j  -  File Repository:"
				+ "Google Code - Subversion:"
				+ "# Non-members may check out a read-only working copy anonymously over HTTP."
				+ "svn checkout http://travellog-aws.googlecode.com/svn/trunk/ travellog-aws-read-only");

		List<ProjectComponents> projectComponents = new ArrayList<ProjectComponents>();

		projectComponents.add(new ProjectComponents("Features/Comments",
				"src/com/amazon/aws/samplecode/travellog/features/comments"));
		projectComponents.add(new ProjectComponents("Features/Journal",
				"src/com/amazon/aws/samplecode/travellog/features/journal"));
		projectComponents.add(new ProjectComponents("Features/Photos",
				"src/com/amazon/aws/samplecode/travellog/features/photos"));
		projectComponents.add(new ProjectComponents("Features/Users",
				"src/com/amazon/aws/samplecode/travellog/features/users"));
		projectComponents.add(new ProjectComponents("Features/Web",
				"src/com/amazon/aws/samplecode/travellog/web"));
		projectComponents.add(new ProjectComponents("Java/Amazon Server",
				"src/com/amazon/aws/samplecode/travellog/aws"));
		projectComponents.add(new ProjectComponents("Java/Database",
				"src/com/amazon/aws/samplecode/travellog/dao"));
		projectComponents.add(new ProjectComponents("Java/Entity",
				"src/com/amazon/aws/samplecode/travellog/entity"));
		projectComponents.add(new ProjectComponents("Java/Utilities",
				"src/com/amazon/aws/samplecode/travellog/util"));
		projectComponents.add(new ProjectComponents("Script Files/CSS",
				"WebContent/css"));
		projectComponents.add(new ProjectComponents("Script Files/JSP",
				"WebContent/WEB-INF/jsp"));
		projectComponents.add(new ProjectComponents("Script Files/JavaScript",
				"WebContent/js"));

		project.setProjectComponents(projectComponents);

		return project;

	}
}
