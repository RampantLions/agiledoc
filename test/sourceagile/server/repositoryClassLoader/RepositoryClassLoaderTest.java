package sourceagile.server.repositoryClassLoader;

import java.io.File;

import junit.framework.TestCase;

import org.approvaltests.Approvals;
import org.approvaltests.UseReporter;

import sourceagile.server.doclet.GetClassDoc;
import sourceagile.shared.Entry;
import sourceagile.shared.Project;
import sourceagile.shared.User;
import sourceagile.shared.data.ProjectsData;
import sourceagile.shared.data.UserData;

import com.spun.util.io.FileUtils;

@UseReporter(ViDiffReporter.class)
public class RepositoryClassLoaderTest extends TestCase {

	public void testFeaturesList() throws Exception {

		User user = UserData.load()[0];
		Project project = ProjectsData.load()[1];
		Entry[] listFeatures = new RepositoryClassLoaderImplementation()
				.ListFeatures(project, user);
		Approvals.approve("Entry", listFeatures);

	}

	public void testGetEntryFeature() throws Exception {

		Entry entryFeature = new Entry();

		File file = new File(
				"test/sourceagile/server/repositoryClassLoader/RepositoryClassLoaderTest.java");

		entryFeature.setTextContent(FileUtils.readFile(file));

		entryFeature.setClassDoc(GetClassDoc.getClassDoc(file));

		Approvals.approve(ClassDocumetationUtilities.toString(entryFeature
				.getClassDoc()));
	}

}
