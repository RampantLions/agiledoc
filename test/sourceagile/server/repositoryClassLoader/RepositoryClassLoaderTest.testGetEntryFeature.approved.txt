public ThisClass extends junit.framework.TestCase
description=
 imports=[java.io.File, junit.framework.TestCase, org.approvaltests.Approvals, org.approvaltests.UseReporter, sourceagile.server.doclet.GetClassDoc, sourceagile.shared.Entry, sourceagile.shared.Project, sourceagile.shared.User, sourceagile.shared.data.ProjectsData, sourceagile.shared.data.UserData, com.spun.util.io.FileUtils]
 -- constructors --
[0] = RepositoryClassLoaderTest()

 -- fields --
.length = 0
 -- methods --
[0] = testFeaturesList()
[1] = testGetEntryFeature()

 todo=false]
